builder/remotecontext/git/gitutils.go

```diff
@@ -19,6 +20,9 @@ type gitRepo struct {
 	subdir string
 }
 
+// nolint: gosimple
+var defaultBranch = regexp.MustCompile("refs\\/heads\\/([^\\s]+)")
```

Comment:
This is probably going to be easier to read/maintain using backticks instead of double quotes (also not sure what the `nolint:` is about?):

```suggestion
var defaultBranch = regexp.MustCompile(`refs/heads/(\S+)`)
```

(Also, I don't think we need to escape `/` since it's not used as a delimiter in Go, and `[^\s]` is the same as `\S`)

Reply:
I picked `nolint:` up from here while working on #42051

https://github.com/moby/moby/blob/a602b052a9c285e9659e9ce007f2aa7f0a73812f/oci/oci.go#L16-L17

Without it, the `gosimple` linter fails, taking the tests along with it. Might have something to do with the double quotes now that you mention it.